app/src/main/java/ee/ioc/phon/android/speechutils/AbstractAudioRecorder.java

```diff
@@ -40,25 +45,34 @@
     // 2 (bytes) * 1 (channels) * 30 (max rec time in seconds) * 44100 (times per second) = 2 646 000 bytes
     // but typically is:
     // 2 (bytes) * 1 (channels) * 20 (max rec time in seconds) * 16000 (times per second) = 640 000 bytes
-    private final byte[] mRecording;
+    final byte[] mRecording;
 
     // TODO: use: mRecording.length instead
     private int mRecordedLength = 0;
+    private AtomicLong mRecordedSessionId = new AtomicLong(0L);
```

Comment:
a session is relevant only in the case of continuous listening during consumption phase (in which we consume the recorded buffer).
As long as the consuming pointer is behind the recording pointer - there's no problem and they are both in the same recording "session". However, if, for any reason, the consuming pointer stalls and the recording pointer passes the consuming pointer, we have a problem with the continuity of the sound since now if the consuming pointer keeps on going it will get data from the far future and not from the near future (as expected). Once the recording pointer passes the consuming pointer (while consumption is in progress) - I'm marking it as a new session. This is of course my idea and is subject to discussion